enos/provider/vbox.py

```diff
@@ -0,0 +1,118 @@
+from provider import Provider
+from jinja2 import Environment, FileSystemLoader
+from ipaddress import IPv4Network
+from ..utils.constants import TEMPLATE_DIR
+from ..utils.extra import build_resources, expand_topology, build_roles
+from itertools import groupby
+# NOTE(msimonin): we should get rid of this
+from execo import Host
+
+import logging
+import vagrant
+import os
+
+SIZES = {
+    'tiny': {
+        'cpu': 1,
+        'mem': 512
+    },
+    'small': {
+        'cpu': 1,
+        'mem': 1024
+    },
+    'medium': {
+        'cpu': 2,
+        'mem': 2048
+    },
+    'big': {
+        'cpu': 3,
+        'mem': 3072,
+    },
+    'large': {
+        'cpu': 4,
+        'mem': 4096
+    }
+}
+
+
+class Vbox(Provider):
+    def init(self, config, calldir, force_deploy=False):
+        """python -m enos.enos up --provider=vbox
+        Read the resources in the configuration files.  Resource claims must be
+        grouped by sizes according to the predefined SIZES map.
+        """
+        self.config = config
+        if 'topology' in self.config:
+            # expand the groups first
+            self.config['topology'] = expand_topology(self.config['topology'])
+            # Build the ressource claim to g5k
+            # We are here using a flat combination of the resource
+            # resulting in (probably) deploying one single region
+            self.config['resources'] = build_resources(self.config['topology'])
+
+        net_pools = {
+            'ip1': list(IPv4Network(u'192.168.142.0/25')),
+            'ip2': list(IPv4Network(u'192.168.143.0/25')),
+            'ip3': list(IPv4Network(u'192.168.144.0/25')),
+        }
+
+        # Build a list of machines that will be used to generate the Vagrantfile
+        self.machines = []
```

Comment:
We don't need class attribute here. Actually, I don't like the idea of using class' attribute, especially because we cannot store the state of a Provider. Going with a class' attribute into the `init` gives you the feeling that you can use this attribute in, for instance, `Vagrant.destroy`, which is wrong.